nibabel/freesurfer/mghformat.py

```diff
@@ -459,9 +459,13 @@ class MGHImage(SpatialImage):
     """ Class for MGH format image
     """
     header_class = MGHHeader
-    files_types = (('image', '.mgh'),)
+    files_types = (('image', '.mgh'),
+                   ('image', '.mgz'))
```

Comment:
Ah - maybe this relates to line 134 of `filename_parser.py`.  Ouch. 

The problem here is that I intended `files_types` to be a list of the types of file that make up a single image.  Here you are re-using it to be a list of types of extensions to try when loading a file.   Did I understand right?